src/parsers/vast.js

```diff
@@ -0,0 +1,149 @@
+import { VASTClient } from 'vast-client'
+
+export default class VASTManager {
+  /**
+   * Initialize a new VastManager instance with one VASTClient lib instance.
+   */
+  constructor() {
+    this.client = new VASTClient()
+  }
+
+  /**
+   * Request VAST XML and returns one VASTResponse.Ad entity.
+   * @param {Object} adData Contains the url to fetch the VAST document.
+   * @returns {Promise} Promise resolved with one VASTResponse.Ad entity or one error.
+   * @see {@link https://github.com/dailymotion/vast-client-js/blob/master/docs/api/class-reference.md#ad}
+   */
+  request(adData) {
+    return adData
+      ? this._processAdData(adData)
+        .then(ads => ads)
```

Comment:
I think we do not need the `then` and `catch` (on the line below) since they are just wrapping the same result returned by `_processAdData` originally.

```javascript
if (!adData) {
  return Promise.reject(...)
}

return this._processAdData(adData)
```

Reply:
Agreed about `then/catch`, but maintain the ternary operator seems to be more meaningful over just omit the `if/else` statement. Another point that I trying to avoid is negated conditions. What do you think about this code below?

```javascript
  request(adData) {
    return adData
      ? this._processAdData(adData)
      : Promise.reject('Invalid adTag received to request VAST')
  }
```

Reply:
Sure! I am used to code in Golang and C#. This way is more idiomatic in Javascript! :D